File: R/linking.R

#' Visualize Linked Data
#'
#' @export
linked_views <- function(polys, dimred, width = NULL, height = NULL, display_opts = list()) {
  x <- list(polys, dimred)
  htmlwidgets::createWidget(
    name = "linked_views",
    x,
    width = width,
    height = height,
    package = "linking"
  )
}

File: inst/htmlwidgets/linking.js
/* Linking Abstract and Physical Maps */

HTMLWidgets.widget({

  name: 'linked_views',
  type: 'output',

  factory: function(el, width, height) {
    return {
      renderValue: function(x) {
	      linked_views(el, width, height, x[0], x[1]);
      },

      resize: function(width, height) {}
    };
  }
});
